### DISCUSSION TITLE
Is it possible to use useInfiniteQuery hook when continuation token for a query with same parameters changes with time?

### DISCUSSION THREAD

**Discussion by rajeshwar904** [TanStack/query#6646] [Q&A]
I have an API which takes query parameters and continuation token as input and returns the set of results and continuation token which can be used to get next set of results. The query results and continuation token can change when the same query with the same continuation token is performed again.

I am trying to implement infinite scrolling using the useInfiniteQuery hook, but when the query is being refetched the data is getting corrupted or some results are duplicated etc. Is there anything that can be done here?

or is it not recommended to use useInfiniteQuery when it is above scenario?

**Comment by TkDodo** [comment]
sorry, the question is too vague for me to answer properly. What does "data is getting corrupted or some results are duplicated etc." mean? The best thing you can do is show a minimal codesandbox reproduction with your problem

**Reply by rajeshwar904** [reply]
![image](https://github.com/TanStack/query/assets/74103263/408b8e1f-5629-4084-aebf-2cc49d0e5ba6)
 The data is fetched as shown above. There is no more data as continuation token is null at the end of 5 calls. Now let us say the refetch happened again,

![image](https://github.com/TanStack/query/assets/74103263/28893729-7fa1-4b1a-8cfd-b992cbedd62b)

Here, After the 5th call the Continuation token is null even then the API call is happening again which is adding duplicate values in the data. The continuation tokens received as part of response here are not same as the ones in the above and the number of calls to load entire data can also be more than or less than or equal to 5 calls.

**Reply by TkDodo** [reply]
I repeat:

> The best thing you can do is show a minimal codesandbox reproduction with your problem
